ocaml/xapi/stream_vdi.ml

```diff
@@ -100,6 +100,53 @@ let write_block ~__context filename buffer ofd len =
     then raise (Api_errors.Server_error (Api_errors.client_error, [ExnHelper.string_of_exn e]))
     else raise e
 
+(* Example = [{"length": 2097152, "flags": 0}, {"length": 1902848, "flags": 3}]*)
+type descriptor = {
+  length : int64;
+  status_flags : int32;
+}
+
+let cycle_descriptors descriptor_list offset =
+  let rec range a b =
+    if a > b then []
+    else a :: range (a + 1) b
```

Comment:
This is not tail recursive

Reply:
Good catch! We should use an accumulator instead.

Reply:
What is the issue with it not being tail recursive?

Reply:
If the list is not processed tail recursively, every element will result in a new function call & stack frame, and we might get a stack overflow for big lists: https://dev.realworldocaml.org/lists-and-patterns.html#scrollNav-5